includes/class-boldgrid-backup-archive-fetcher.php

```diff
@@ -0,0 +1,290 @@
+	 * @var string
+	 */
+	private $filepath;
+
+	/**
+	 * Fetcher info.
+	 *
+	 * After a successful fetch, this array will have info about our new backup.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var array {
+	 *     @type string $filepath        The filepath to the archive.
+	 *     @type string $detailsUrl      The admin url to the details page for this archive.
+	 *     @type string $archiveFilename The filename of the archive.
+	 *     @type int    $archiveKey      The archive key.
+	 * }
+	 */
+	private $info = [];
+
+	/**
+	 * The path to the archive's log.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var string
+	 */
+	private $log_filepath;
+
+	/**
+	 * The response received when trying to download the file.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var mixed
+	 */
+	private $response;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param string $url The url we will be downloading.
+	 */
+	public function __construct( $url ) {
+		$this->core = apply_filters( 'boldgrid_backup_get_core', null );
+
+		$this->url = $url;
+	}
+
+	/**
+	 * Download a backup file from a remote server.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return bool True on success.
+	 */
+	public function download() {
+		if ( ! $this->is_valid_url() ) {
+			$this->error = __( 'Invalid URL address.', 'boldgrid-backup' );
+			return false;
+		}
+
+		if ( ! $this->is_valid_backupdir() ) {
+			$this->error = implode( '<br />', $this->core->backup_dir->errors );
+			return false;
+		}
+
+		$this->filepath = $this->core->upload->get_save_path( basename( $this->url ) );
+
+		$this->response = wp_remote_get(
+			$this->url,
+			[
+				'filename'  => $this->filepath,
+				'headers'   => 'Accept: ' . implode( ', ', $this->allowed_content_types ),
+				'sslverify' => false,
+				'stream'    => true,
+				'timeout'   => MINUTE_IN_SECONDS * 20,
+			]
+		);
+
+		if ( $this->is_successful_download() ) {
+			$this->post_successful_download();
+
+			return true;
+		} else {
+			$this->core->wp_filesystem->delete( $this->filepath );
+
+			$this->error = __(
+				'Could not retrieve the remote file.  It may not be a ZIP file, or the link is no longer valid.',
+				'boldgrid-backup'
+			);
+
+			return false;
+		}
+	}
+
+	/**
+	 * Get our error message.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return string
+	 */
+	public function get_error() {
+		return $this->error;
+	}
+
+	/**
+	 * Get our array of info after a successful download.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return array
+	 */
+	public function get_info() {
+		return $this->info;
+	}
+
+	/**
+	 * Whether or not we encountered an error during the download process.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return bool
+	 */
+	public function has_error() {
+		return ! empty( $this->error );
+	}
+
+	/**
+	 * Whether or not the call to download the file was successful.
+	 *
+	 * This does not represent the success of the download() method, but instead the state of the
```

Comment:
If the purpose of this is not call to download was successful or not, rather than whether the download itself was successful, maybe the method should be renamed. maybe 'is_call_successful()'

Reply:
https://github.com/BoldGrid/boldgrid-backup/commit/e6d449a6f9d40f5c1e5063b58b0fc861460f8c9a